### PR TITLE
Subcommands galore! Also better specified-flags.

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

File: server/cli/accounting.go [hunk 1]
```diff
@@ -105,7 +105,7 @@ func runRmAcct(cmd *cobra.Command, args []string) {
 // A setAcctCmd command creates a new or updates an existing acct
 // config.
 var setAcctCmd = &cobra.Command{
-	Use:   "set-acct [options] <key-prefix> <acct-config-file>",
+	Use:   "set [options] <key-prefix> <acct-config-file>",
 	Short: "create or update an accounting config for key prefix\n",
```
Comment (petermattis) [inline server/cli/accounting.go]: FYI, we were using `\n` at the end of the `Short` string to create groupings. I think these can be removed now.

File: server/cli/flags.go [hunk 2]
```diff
@@ -20,73 +20,87 @@ package cli
...
-		"caches, shared evenly if there are multiple storage devices.")
-
-	pflag.DurationVar(&ctx.ScanInterval, "scan-interval", ctx.ScanInterval, "specify "+
-		"--scan_interval to adjust the target for the duration of a single scan "+
-		"through a store's ranges. The scan is slowed as necessary to approximately"+
-		"achieve this duration.")
+	for _, cmd := range nodeCmds {
+		// Server flags.
+
+		cmd.Flags().StringVar(&ctx.Addr, "addr", ctx.Addr, "the host:port to bind for HTTP/RPC traffic")
```
Comment (petermattis) [inline server/cli/flags.go]: Add `f := cmd.Flags()` and then `s/cmd.Flags()/f/g` below.

**Comment by petermattis** [comment]
None of the `server/cli` tests needed updating?

**Comment by tamird** [comment]
@petermattis they did :smiling_imp:

File: server/cli/flags.go (hunk 2 above)
Comment (tamird) [inline server/cli/flags.go]: :+1:

File: server/cli/accounting.go (hunk 1 above)
Comment (tamird) [inline server/cli/accounting.go]: removed some - kept the ones on quit and create-node-cert

File: server/cli/flags.go [hunk 3]
```diff
@@ -20,73 +20,87 @@ package cli
...
+
+		cmd.Flags().BoolVar(&ctx.Linearizable, "linearizable", ctx.Linearizable, "enables linearizable behaviour "+
+			"of operations on this node by making sure that no commit timestamp is reported "+
+			"back to the client until all other node clocks have necessarily passed it.")
+
+		cmd.Flags().StringVar(&ctx.Stores, "stores", ctx.Stores, "specify a comma-separated list of stores, "+
```
Comment (mrtracy) [inline server/cli/flags.go]: You're going to get some merge conflicts with my latest change, assuming I check it in first - I added a server var, but also standardized all of these help messages to use a single hyphen in examples instead a double hyphen.

Comment (tamird) [inline server/cli/flags.go]: single hyphens are no longer valid, though

File: server/cli/accounting.go [hunk 4]
```diff
@@ -137,3 +137,18 @@ func runSetAcct(cmd *cobra.Command, args []string) {
...
+
+var acctCmds = []*cobra.Command{
+	getAcctCmd,
+	lsAcctsCmd,
+	rmAcctCmd,
+	setAcctCmd,
+}
+
+var acctCmd = &cobra.Command{
+	Use: "account",
```
Comment (spencerkimball) [inline server/cli/accounting.go]: I wonder if we should be explicit and call this accounting. "account" is maybe a little misleading or confusing.

Comment (tamird) [inline server/cli/accounting.go]: done

File: server/cli/flags.go [hunk 5]
```diff
@@ -20,73 +20,93 @@ package cli
...
+			"approximate interval (time.Duration) for gossiping new information to peers.")
+
+		// Engine flags.
+
+		f.Int64Var(&ctx.CacheSize, "cache-size", ctx.CacheSize, "total size in bytes for "+
+			"caches, shared evenly if there are multiple storage devices.")
+
+		f.DurationVar(&ctx.ScanInterval, "scan-interval", ctx.ScanInterval, "specify "+
+			"--scan_interval to adjust the target for the duration of a single scan "+
+			"through a store's ranges. The scan is slowed as necessary to approximately"+
```
Comment (spencerkimball) [inline server/cli/flags.go]: missing a space after "approximately"

File: server/cli/flags.go [hunk 6]
```diff
@@ -20,73 +20,93 @@ package cli
...
+	clientCmds = append(clientCmds, rangeCmds...)
+	clientCmds = append(clientCmds, acctCmds...)
+	clientCmds = append(clientCmds, permCmds...)
+	clientCmds = append(clientCmds, zoneCmds...)
+
+	for _, cmd := range clientCmds {
+		cmd.Flags().StringVar(&ctx.Addr, "addr", ctx.Addr, "the address for connection to the cockroach cluster.")
+	}
+
+	for _, cmd := range append(nodeCmds, clientCmds...) {
```
Comment (petermattis) [inline server/cli/flags.go]: Not sure if the `append` is doing what you want. Seems suspicious to be appending to `nodeCmds` here. I think more idiomatic go would be to do something like:

```
  for _, cmds := range [][]*cobra.Command{nodeCmds, clientCmds} {
    for _, cmd := range cmds {
    }
  }
```

**Comment by petermattis** [comment]
LGTM

**Comment by spencerkimball** [comment]
Nice. LGTM

**Pull request merged**